<launch>
  <arg name="home_dir"          default="$(find pointcloud_processing_server)" />
  <arg name="rviz_file"         default="$(arg home_dir)/rviz/camera_only.rviz" />  
  <arg name="camera_type"       default="r200" />
  <arg name="launch_seg"        default="true" />
  <arg name="yaml_file_name"    default="basic" />
  <arg name="prim_search_name"  default="tabletop_binder_bottle" />

  <!-- RViz -->
  <!--node
    name    = "rviz_node"	
    pkg     = "rviz"
    type    = "rviz"
    args    = "-d $(arg rviz_file) -s $(arg home_dir)/rviz/rviz_splash.png"
    output  = "screen"
  >
  </node -->

  <!-- Map Publisher -->
  <rosparam command="load" file="$(find pointcloud_processing_server)/map_maker/map_maker_pcs.yaml" />
  <node
    name    = "map_maker_pcs"
    pkg     = "pointcloud_processing_server"
    type    = "map_maker_pcs"
    args    = ""
    output  = "screen"
  >
  </node>
    
  <!-- Camera -->
  <include file="$(find realsense_camera)/launch/$(arg camera_type)_nodelet_rgbd.launch">
    <!--arg name="base_frame_id" value="camera_link" /-->
  </include>

  <rosparam command="load" file="$(find pointcloud_processing_server)/pointcloud_process_from_yaml/$(arg yaml_file_name).yaml" />
  <rosparam command="load" file="$(find pointcloud_processing_server)/primitive_search/$(arg prim_search_name).yaml" />
  <group if="$(arg launch_seg)">
    <!-- PointcloudProcessing Server -->
    <node
      name    = "pointcloud_processing_server"
      pkg     = "pointcloud_processing_server"
      type    = "pointcloud_processing_server"
      args    = ""
      output  = "screen"
    >
    </node>
    <!-- PointcloudProcessing Client -->
    <node
      name    = "test_segmentation"
      pkg     = "pointcloud_processing_server"
      type    = "example_client"
      args    = ""
      output  = "screen"
    >
      <param name="base_yaml_file_name" type="str" value="$(arg yaml_file_name)" />
    </node>
    <!-- Primitive Search Server -->
    <node 
      name    = "primitive_search"
      pkg     = "pointcloud_processing_server"
      type    = "primitive_search" 
      args    = ""
      output  = "screen"
    >
    </node>

  </group>

</launch>
